src/main/java/io/lettuce/core/protocol/SharedLock.java

```diff
@@ -125,6 +129,11 @@ private void lockWritersExclusive() {
                     exclusiveLockOwner = Thread.currentThread();
                     return;
                 }
+                // reentrant exclusive lock
+                if (WRITERS.compareAndSet(this, sharedCnt.get(), -1)) {
```

Comment:
The previous `if` statement becomes redundant and we can remove it.
- if the current thread has no writer subscriptions then the `sharedCnt.get()` is 0 and it repeats the same logic
- if the current thread has one or more writer subscriptions we can check it with the same check

In short - IMHO we can remove lines 128 to 131

Reply:
thanks, those suggestions have been applied.